### PR TITLE
dockerfile: use rockylinux

### LINKED ISSUE #38933: use rockylinux as base image
## Feature Request

**Is your feature request related to a problem? Please describe:**
CentOS is dead

**Describe the feature you'd like:**

Use Rockylinux as base image, please provide Dockfile, and list all the package(and version) it depends.

**Describe alternatives you've considered:**

no choice

**Teachability, Documentation, Adoption, Migration Strategy:**

https://pingcap.feishu.cn/wiki/wikcnRuB2X25Pb6O5w29GqlEnZg

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #38933

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
```make docker```
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -13,16 +13,24 @@
...
+    && curl -fsSL "$GOLANG_DOWNLOAD_URL" -o golang.tar.gz \
+	&& tar -C /usr/local -xzf golang.tar.gz \
+	&& rm golang.tar.gz
 
 COPY . /tidb
 ARG GOPROXY
 RUN export GOPROXY=${GOPROXY} && cd /tidb && make server
 
-FROM alpine:latest
+FROM rockylinux:9
```
Comment (xhebox) [inline Dockerfile:33]: Maybe use the `9-minimal`? It is clearly larger than the original alpine images.

File: Dockerfile [hunk 2]
```diff
@@ -13,16 +13,24 @@
...
+ENV GOLANG_VERSION 1.19.3
+ENV ARCH amd64
+ENV GOLANG_DOWNLOAD_URL https://dl.google.com/go/go$GOLANG_VERSION.linux-$ARCH.tar.gz
+ENV GOPATH /go
+ENV GOROOT /usr/local/go
+ENV PATH $GOPATH/bin:$GOROOT/bin:$PATH
+RUN yum update -y && yum groupinstall 'Development Tools' -y \
+    && curl -fsSL "$GOLANG_DOWNLOAD_URL" -o golang.tar.gz \
+	&& tar -C /usr/local -xzf golang.tar.gz \
+	&& rm golang.tar.gz
```
[the remark this answers is not in the document]
Comment (hawkingrei) [reply Dockerfile:27]: In fact, I cannot find an official yum repository to install the latest golang so I have to install golang by hand.

**Review by xhebox — COMMENTED** [review]
LGTM, but one thing, releasing is not using this file. I don't think changing this makes any difference other than using glibc.

File: Dockerfile (hunk 1 above)
Comment (hawkingrei) [reply Dockerfile]: Done

**Comment by hawkingrei** [comment]
> LGTM, but one thing, releasing is not using this file. I don't think changing this makes any difference other than using glibc.

I agree with you. But it might be some compliance issue.

**Review by xhebox — APPROVED** [review]

**Comment by xhebox** [comment]
> > LGTM, but one thing, releasing is not using this file. I don't think changing this makes any difference other than using glibc.
> 
> I agree with you. But it might be some compliance issue.

Clearly, that is not a problem. Because I am using musl. If there is any problem, I will fix it.  😄 

And, go does not have much left problems on musl libc. Talking about containers for go programs, alpine is the best choice.

/merge

**Pull request merged**